qt/python/mantidqt/mantidqt/widgets/workspacedisplay/matrix/delegate.py

```diff
@@ -44,10 +44,16 @@ def paint(self, painter: QPainter, option: QStyleOptionViewItem, index: QModelIn
         try:
             painter.setClipRect(opt.rect)
             foreground_colour, background_colour = index.data(Qt.ForegroundRole), index.data(Qt.BackgroundRole)
-            if foreground_colour is not None:
-                painter.setPen(foreground_colour)
-            if background_colour is not None:
-                painter.fillRect(option.rect, background_colour)
+            state_selected = option.state & QStyle.State_Selected
```

Comment:
In the tests `QStyle.State_Selected` is always interpreted here as `true` - is there a situation where it could be false that we also need to test?

Reply:
`QStyle.State_Selected` is effectively `false` by default, so the two preceding tests `test_custom_eliding_delegate_respects_background_color` and `test_custom_eliding_delegate_respects_foreground_color` cover this.

Reply:
On debug it looks like even for those tests `bool(QStyle.State_Selected) == true` unless I've missed something?

Reply:
OK I see, so it isn't cast to a bool before the `&` - looking at the code it does seem obvious that it's a comparison, I guess that's also why you created the variable (rather than just write it with the if statement)? Sorry not seen these bitwise operations before, will approve!

Reply:
I too have only come across them seldomly, I had to refer to this when writing the code https://stackoverflow.com/questions/73395164/how-do-i-check-styleoption-state-in-pyside6